llvm/lib/IR/Verifier.cpp

```diff
@@ -2377,6 +2377,32 @@ void Verifier::verifyFunctionAttrs(FunctionType *FT, AttributeList Attrs,
       CheckFailed("'allockind()' can't be both zeroed and uninitialized");
   }
 
+  if (Attrs.hasFnAttr("alloc-variant-zeroed")) {
+    if (auto A = Attrs.getFnAttr("alloc-variant-zeroed"); A.isValid()) {
+      StringRef s = A.getValueAsString();
+      Check(!s.empty(), "\"alloc-variant-zeroed\" must not be empty");
+      if (auto *F = dyn_cast<Function>(V)) {
```

Comment:
We should also verify that F and the Variant have the same function type (i.e. signature).